packages/core/src/payment/strategies/worldpayaccess/worldpayaccess-payment-strategy.ts

```diff
@@ -186,4 +211,39 @@ export default class WorldpayaccessPaymetStrategy extends CreditCardPaymentStrat
 
         return iframe;
     }
+
+    private _submitAdditionalAction() {
+        if (this._shouldRenderHostedForm) {
+            if (!this._hostedForm || !this._hostedForm.submit) {
+                throw new NotInitializedError(
+                    NotInitializedErrorType.PaymentNotInitialized
+                );
+            }
+
+            const hostedForm = this._hostedForm;
+
+            return async (paymentPayload: OrderPaymentRequestBody) => {
+                await hostedForm.submit(paymentPayload);
+                return this._store.getState();
+            }
+        } else {
+            return async (paymentPayload: OrderPaymentRequestBody) => {
+                return await this._store.dispatch(this._paymentActionCreator.submitPayment( paymentPayload ));
+            }
+        }
+    }
+
+    private _isValidJson(str:string) {
```

Comment:
Can we name this method appropriately, seems like we are not validating JSON but if it's a json with a `sessionId` key

Reply:
both if `JSON.parse(str)` fails in the convertion then catch is going  to return `false` but if it's a valid json we need to ensure that it's a json with `sessionId`

Reply:
Yes, I was suggesting to name it keeping the sessionId in mind. Since by the looks of it we it needs to be a valid JSON with a key of sessionId